File: tests/fixtures/check_then_act/ts/bare_write_fixture.ts
// A helper named like a write is a write even when it is not a method.
export async function order(repo: any, id: string) {
  const row = repo.get(id);
  if (row) return; // expect: check
  await repo.sync(id); // expect: gap await
  saveOrder(id); // expect: write
}

declare function saveOrder(id: string): void;

File: tests/fixtures/check_then_act/py/bare_write_fixture.py
# A helper named like a write is a write even when it is not a method.
async def order(repo, order_id):
    row = repo.get(order_id)
    if row:  # expect: check
        return
    await repo.sync(order_id)  # expect: gap await
    save_order(order_id)  # expect: write


def save_order(order_id):
    pass
